types/events.go

```diff
@@ -39,6 +44,97 @@ func (em EventManager) ABCIEvents() []abci.Event {
 	return em.events.ToABCIEvents()
 }
 
+// EmitTypedEvent takes typed event and emits converting it into Event
+func (em *EventManager) EmitTypedEvent(tev proto.Message) error {
+	event, err := TypedEventToEvent(tev)
+	if err != nil {
+		return err
+	}
+
+	em.EmitEvent(event)
```

Comment:
If we want to force the usage of TypedEvents everywhere, maybe `EmitEvent` and `EmitEvents` could both be marked as deprecated in this PR?

Once https://github.com/cosmos/cosmos-sdk/issues/7563 is finished with all modules, we can even make these private.